api/application/client.go

```diff
@@ -497,7 +497,11 @@ func (c *Client) SetRelationStatus(relationId int, status relation.Status) error
 	if err := c.facade.FacadeCall("SetRelationStatus", args, &results); err != nil {
 		return errors.Trace(err)
 	}
-	return results.OneError()
+	err := results.OneError()
+	if status == relation.Joined && params.ErrCode(err) == params.CodeForbidden {
+		return errors.New("cannot resume a relation to an offer where the offer user doesn't have consume access")
```

Comment:
why not return that message from the server? this feels like guessing. otherwise use a more specific error code?

Reply:
The server talks about setting status to "joined" which is what happens under the covers when a relation is resumed. But from the CLI perspective, that's an internal detail.

Reply:
I could use a new error code instead.